oxide-vpc/tests/common/icmp.rs

```diff
@@ -101,12 +109,40 @@ pub fn gen_icmp_echo(
         &EtherMeta { dst: eth_dst, src: eth_src, ether_type: EtherType::Ipv4 };
 
     let total_len = EtherHdr::SIZE + ip4.hdr_len() + icmp.buffer_len();
-    let mut pkt = Packet::alloc_and_expand(total_len);
-    let mut wtr = pkt.seg0_wtr();
-    eth.emit(wtr.slice_mut(EtherHdr::SIZE).unwrap());
-    ip4.emit(wtr.slice_mut(ip4.hdr_len()).unwrap());
-    wtr.write(&icmp_bytes).unwrap();
-    pkt.parse(Out, VpcParser::new()).unwrap()
+
+    if !multi_seg {
+        let mut pkt = Packet::alloc_and_expand(total_len);
+        let mut wtr = pkt.seg0_wtr();
+        eth.emit(wtr.slice_mut(EtherHdr::SIZE).unwrap());
+        ip4.emit(wtr.slice_mut(ip4.hdr_len()).unwrap());
+        wtr.write(&icmp_bytes).unwrap();
+        pkt.parse(Out, VpcParser::new()).unwrap()
+    } else {
+        let mut pkt = Packet::alloc_and_expand(EtherHdr::SIZE);
+        let mut wtr = pkt.seg_wtr(0);
+        eth.emit(wtr.slice_mut(EtherHdr::SIZE).unwrap());
+
+        /*
+         * TODO
+         * It seems like this should also work, but it does not. Leaving this as
```

Comment:
I think this is because of the missing `body.seg_offset > 0` check I mentioned above.